File: JavaScript/e030.js
function sumpow(num, p = 5) {
    const digits = Array.from(num.toString());
    let sum = 0;
    digits.forEach(x => {
        sum += x ** p;
    })
    return sum;
}
let sum = 0;
for (let i = 4000; i <= 200000; i++) {
    if (i === sumpow(i)) {
        sum += i;
    }
}
console.log(sum);

File: e030.py
"""
Problem 30: Digit fifth powers
Find the sum of all the numbers that can be written as the
sum of fifth powers of their digits.
"""


def sumpow(num, p=5):
    digits = [int(i) for i in str(num)]
    return sum([x ** p for x in digits])


s = 0
for i in range(4000, 200000):
    if i == sumpow(i):
        s += i
print(s)
